### ISSUE TITLE
StronglyConnectedComponentsFast missing sccs

### ISSUE THREAD

**Issue by stefandollase** [Sable/soot#836]
`StronglyConnectedComponentsFast` does not seem to identify any sccs if the given `DirectedGraph` does not have any heads.

```
MutableDirectedGraph<String> graph = new HashMutableDirectedGraph<String>();
graph.addNode("foo");
graph.addNode("bar");
graph.addEdge("foo", "foo");
graph.addEdge("foo", "bar");
System.out.println(new StronglyConnectedComponentsFast<>(graph).getComponents());
```

This outputs an empty list: `[]`

I would have expected an output like `[[foo], [bar]]`.

**Comment by mbenz89** [comment]
I would be happy to merge a fix if you can provide one.

Manuel

**Comment by mbenz89** [comment]
Fixed in #838
